script.js

```diff
@@ -1,4 +1,593 @@
+  if (currentInitialScore > 21) {
+    currentFinalScore = currentInitialScore;
+    var j = 0;
+    while (j < sortedCardArray.length) {
+      if (sortedCardArray[j].rank == 11 && currentFinalScore > 21) {
+        currentFinalScore -= 10;
+        j += 1;
+      } else if (sortedCardArray[j].rank < 11) {
+        currentFinalScore = currentInitialScore;
+      }
+      return currentFinalScore;
+    }
+  } else return currentInitialScore;
+};
+
+//FUNCTION 5 - Create loop to provide the result sentence for n number of cards drawn. To display each card's attributes within the array.
+var outputResult = function (cardArray) {
+  var output = "";
+  var k = 0;
+  while (k < cardArray.length) {
+    if (k == cardArray.length - 1) {
+      output += `and ${cardArray[k].name} ${cardArray[k].emoji}`;
+    } else output += `${cardArray[k].name} ${cardArray[k].emoji}, `;
+
+    k += 1;
+  }
+  return output;
+};
+
+//FUNCTION 6 FOR MULTIPLAYER - Create loop to show all players' final hands at the end of the round when win/lose/tie is determined.
+//Info taken from global variable allplayercard array.
+//Outer loop is for player 1 to n.
+//Inner loop is for cards 1 to n for each player.
+//Extract out attributes of each card as result string.
+var showAllPlayerHands = function (numberOfPlayers) {
+  var playerCounter = 0;
+  var result = "";
+  while (playerCounter < numberOfPlayers) {
+    var innerCardCounter = 0;
+    while (innerCardCounter < allPlayerCardArray[playerCounter].length) {
+      if (innerCardCounter == 0) {
+        result +=
+          "Player " +
+          String(Number(playerCounter + 1)) +
+          ": " +
+          String(allPlayerCardArray[playerCounter][innerCardCounter].name) +
+          String(allPlayerCardArray[playerCounter][innerCardCounter].emoji) +
+          ", ";
+      } else if (
+        innerCardCounter + 1 ==
+        allPlayerCardArray[playerCounter].length
+      ) {
+        result +=
+          "& " +
+          String(allPlayerCardArray[playerCounter][innerCardCounter].name) +
+          String(allPlayerCardArray[playerCounter][innerCardCounter].emoji) +
+          ".<br>";
+      } else {
+        result +=
+          String(allPlayerCardArray[playerCounter][innerCardCounter].name) +
+          String(allPlayerCardArray[playerCounter][innerCardCounter].emoji) +
+          ", ";
+      }
+      innerCardCounter += 1;
+    }
+    playerCounter += 1;
+  }
+  return result;
+};
+
+//FUNCTION 7 FOR MULTIPLAYER BETTING - Create loop to show all players' coins at the end of the round when win/lose/tie is determined.
+var showAllPlayerCoins = function (numberOfPlayers) {
+  var playerCounter = 0;
+  var result = "";
+  while (playerCounter < numberOfPlayers) {
+    result +=
+      "Player " +
+      String(Number(playerCounter + 1)) +
+      ": " +
+      allPlayerCoins[playerCounter] +
+      " coins <br>";
+    playerCounter += 1;
+  }
+  return result;
+};
+
+//GLOBAL VARIABLES.
```

Comment:
by convention, global variables are usually right at the top of this file. But in any case, it's good that you grouped all of them together. i also like how you kept images as variables for easy use